src/mpid/ch4/shm/posix/posix_am_impl.h

```diff
@@ -48,12 +48,15 @@ static inline int MPIDI_POSIX_am_init_req_hdr(const void *am_hdr,
     MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_POSIX_AM_INIT_REQ_HDR);
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_POSIX_AM_INIT_REQ_HDR);
 
+    /* XXX - I don't know why this inline is here. */
```

Comment:
The POSIX_AM_REQUEST_INLINE symbol turns on a feature which allowed lightweight (fast) allocation of the "req_hdr" object instead of expensive allocation one via MPIDI_CH4R_get_buf.

Usage of MPIDI_CH4R_get_buf has bad effect on message rate performance.

Reply:
But this only seems to be set if you do it via `-D` at compile time, correct?

Reply:
Would it make sense to make this a configure flag?

Reply:
Yes, it is correct. If this feature considered helpful, it can be turned on by adding "-D POSIX_AM_REQUEST_INLINE" in the CFLAGS list.